src/extensions/default/DebugCommands/main.js

```diff
@@ -257,6 +273,485 @@ define(function (require, exports, module) {
+                } else {
+                    _prefVar = prefObj.keys;
+                }
+
+                if (Array.isArray(_prefVar)) {
+                    // In cases of array the
+                    // typeof is returning a function.
+                    prefType = "array";
+                }
+
+            } else {
+                prefType = typeof (prefObj);
+            }
+        }
+
+        // Now make sure we recognize this format.
+        if (!_isSupportedPrefType(prefType)) {
+            prefType = "undefined";
+        }
+
+        return prefType;
+    }
+
+    function _isValidPref(pref) {
+
+        // Make sure to generate pref description only for
+        // user overrides and don't generate for properties
+        // meant to be used for internal purposes. Also check
+        // if the preference type is valid or not.
+        if (pref && !pref.excludeFromHints && _getObjType(pref) !== "undefined") {
+            return true;
+        }
+
+        return false;
+    }
+
+   /*
+    * This method tries to matach between initial objects
+    * and key objects and then aggregates objects from both
+    * the properties.
+    */
+    function _getObjKeys(prefObj) {
+
+        var finalObj = {},
+            property,
+            keysFound = false;
+
+        if (!prefObj) {
+            return {};
+        }
+
+        if (typeof (prefObj.initial) === "object") {
+            // iterate through the list.
+            keysFound = true;
+            for (property in prefObj.initial) {
+                if (prefObj.initial.hasOwnProperty(property)) {
+                    finalObj[property] = prefObj.initial[property];
+                }
+            }
+        }
+
+        if (typeof (prefObj.keys) === "object") {
+            // iterate through the list.
+            var allKeys = prefObj.keys;
+            keysFound = true;
+            for (property in allKeys) {
+                if (allKeys.hasOwnProperty(property)) {
+                    finalObj[property] = prefObj.keys[property];
+                }
+            }
+        }
+
+        // Last resort: Maybe plain objects, in which case
+        // we blindly extract all the properties.
+        if (keysFound === false) {
+            for (property in prefObj) {
+                if (prefObj.hasOwnProperty(property)) {
+                    finalObj[property] = prefObj[property];
+                }
+            }
+        }
+
+        return finalObj;
+    }
+
+    function _formatDefault(prefObj, prefName, tabIndentStr) {
+
+        if (!prefObj || typeof (prefName) !== "string" || _getObjType(prefObj) === "object") {
+            // return empty string in case of
+            // object or pref is not defined.
+            return "";
+        }
+
+        var prefDescription = prefObj.description || "",
+            prefDefault     = prefObj.initial,
+            prefFormatText  = tabIndentStr + "\t// {0}\n" + tabIndentStr + "\t\"{1}\": {2}",
```

Comment:
This function is responsible for formatting the code it generates. That, basically, is the principle I have used. For first level preferences like `closeBrackets: false`, `tabIndentStr` is going to be empty.  Consider a preference which has nested tags.

```
"closeTags": {
		// Close when > of opening tag is typed
		"whenOpening": false,

		// Close when / of closing tag is typed
		"whenClosing": false,

		// An array of tags that when opened has a blank line
		"indentTags": [],

		// An array of tags that should not be auto-closed
		"dontCloseTags": []
	},

```

When the same function is called for `"whenOpening: false"`, that is when `tabIndentStr` has some value.